File: py-appscript/trunk/Tests/test_osax.py
#!/usr/local/bin/python

import unittest, commands
import osax, mactypes, aem

class TC_OSAX(unittest.TestCase):
	
	def test_1(self):
		sa = osax.ScriptingAddition('Standardadditions')
		
		self.assertEqual(65, sa.ASCII_number('A'))
		
		self.assertEqual(mactypes.Alias("/Applications/"), sa.path_to(osax.k.applications_folder))
		
		self.assertEqual(mactypes.Alias("/Library/Scripts/"), 
				sa.path_to(osax.k.scripts_folder, from_=osax.k.local_domain))
		
		self.assertRaises(RuntimeError, getattr, sa, 'non_existent_command')
	
	
	def test_2(self):
		sa = osax.ScriptingAddition('Standardadditions', name='Finder')
		self.assertEqual(65, sa.ASCII_number('A'))
		self.assertEqual(mactypes.Alias("/System/Library/CoreServices/Finder.app/"), sa.path_to(None))
	
	
	def test_3(self):
		sa = osax.ScriptingAddition('Standardadditions', creator='MACS')
		self.assertEqual(65, sa.ASCII_number('A'))
		self.assertEqual(mactypes.Alias("/System/Library/CoreServices/Finder.app/"), sa.path_to(None))
	
	
	def test_4(self):
		sa = osax.ScriptingAddition('Standardadditions', id='com.apple.finder')
		self.assertEqual(65, sa.ASCII_number('A'))
		self.assertEqual(mactypes.Alias("/System/Library/CoreServices/Finder.app/"), sa.path_to(None))
	
	
	def test_5(self):
		pid = int(commands.getoutput("top -l1 | grep Finder | awk '{ print $1 }'"))
		sa = osax.ScriptingAddition('Standardadditions', pid=pid)
		self.assertEqual(mactypes.Alias("/System/Library/CoreServices/Finder.app/"), sa.path_to(None))
		self.assertEqual(65, sa.ASCII_number('A'))
	

	def test_6(self):
		sa = osax.ScriptingAddition('Standardadditions', aemapp=aem.Application("/System/Library/CoreServices/Finder.app/"))
		self.assertEqual(65, sa.ASCII_number('A'))
		self.assertEqual(mactypes.Alias("/System/Library/CoreServices/Finder.app/"), sa.path_to(None))
	


if __name__ == '__main__':
	unittest.main()

File: rb-appscript/tags/rb-appscript-0.5.3/test/test_osax.rb
#!/usr/bin/ruby -w

begin; require 'rubygems'; rescue LoadError; end

require 'test/unit'
require 'osax'

class TC_OSAX < Test::Unit::TestCase
	
	def test_1
		sa = OSAX.osax('Standardadditions')
		
		assert_equal(65, sa.ASCII_number('A'))
		
		assert_equal(MacTypes::Alias.path("/Applications/"), sa.path_to(:applications_folder))
		
		assert_equal(MacTypes::Alias.path("/Library/Scripts/"), 
				sa.path_to(:scripts_folder, :from=>:local_domain))
		
		assert_raises(RuntimeError) { sa.non_existent_command }
	end
	
	def test_2
		sa = OSAX.osax('Standardadditions').by_name('Finder')
		assert_equal(65, sa.ASCII_number('A'))
		assert_equal(MacTypes::Alias.path("/System/Library/CoreServices/Finder.app/"), sa.path_to(nil))
	end
	
	def test_3
		sa = OSAX.osax('Standardadditions').by_creator('MACS')
		assert_equal(65, sa.ASCII_number('A'))
		assert_equal(MacTypes::Alias.path("/System/Library/CoreServices/Finder.app/"), sa.path_to(nil))
	end
	
	def test_4
		sa = OSAX.osax('Standardadditions').by_id('com.apple.finder')
		assert_equal(65, sa.ASCII_number('A'))
		assert_equal(MacTypes::Alias.path("/System/Library/CoreServices/Finder.app/"), sa.path_to(nil))
	end
	
	def test_5
		sa = OSAX.osax('Standardadditions').by_pid(`top -l1 | grep Finder | awk '{ print $1 }'`.to_i)
		assert_equal(MacTypes::Alias.path("/System/Library/CoreServices/Finder.app/"), sa.path_to(nil))
		assert_equal(65, sa.ASCII_number('A'))
	end

	def test_6
		sa = OSAX.osax('Standardadditions').by_aem_app(AEM::Application.by_path("/System/Library/CoreServices/Finder.app/"))
		assert_equal(65, sa.ASCII_number('A'))
		assert_equal(MacTypes::Alias.path("/System/Library/CoreServices/Finder.app/"), sa.path_to(nil))
	end


end